sdk/keyvault/Azure.Security.KeyVault.Administration/src/KeyVaultBackupClient.cs

```diff
@@ -444,6 +536,87 @@ internal virtual Response<FullBackupDetailsInternal> GetBackupDetails(string job
+
+                // Rest client returns an Operation without headers, so we need to create a new response with headers.
+                var headers = new AzureSecurityKeyVaultAdministrationFullBackupHeaders(operation.GetRawResponse());
+                var responseWithHeaders = ResponseWithHeaders.FromValue(headers,operation.GetRawResponse());
+
+                return new KeyVaultBackupOperation(this, responseWithHeaders);
+            }
+            catch (Exception ex)
+            {
+                scope.Failed(ex);
+                throw;
+            }
+         }
+
+         /// <summary>
+         /// Initiates a pre-backup check on the Key Vault. This operation checks if it is possible to back up the entire collection of keys from a Key Vault.
+         /// </summary>
+         /// <param name="blobStorageUri">The <see cref="Uri"/> for the blob storage resource.</param>
+         /// <param name="sasToken">Optional Shared Access Signature (SAS) token to authorize access to the blob. If null, Managed Identity will be used to authenticate instead.</param>
+         /// <param name="cancellationToken">A <see cref="CancellationToken"/> controlling the request lifetime.</param>
+         /// <exception cref="ArgumentNullException"><paramref name="blobStorageUri"/> or <paramref name="sasToken"/> is null.</exception>
+         /// <exception cref="RequestFailedException">The server returned an error. See <see cref="Exception.Message"/> for details returned from the server.</exception>
+         /// <returns>A <see cref="KeyVaultBackupOperation"/> representing the result of the operation.</returns>
+         public virtual KeyVaultBackupOperation StartPreBackup(Uri blobStorageUri, string sasToken = default, CancellationToken cancellationToken = default)
+         {
+             Argument.AssertNotNull(blobStorageUri, nameof(blobStorageUri));
+             Argument.AssertNotNull(sasToken, nameof(sasToken));
+
+             using DiagnosticScope scope = _diagnostics.CreateScope($"{nameof(KeyVaultBackupClient)}.{nameof(StartPreBackup)}");
+             scope.Start();
+             try
+             {
+                Operation<FullBackupDetailsInternal> operation = _restClient.PreFullBackup(
+                     WaitUntil.Started,
+                     new PreBackupOperationParameters(
+                         blobStorageUri.AbsoluteUri,
+                         sasToken,
+                         useManagedIdentity: sasToken == null
+                         ),
+                     cancellationToken);
+
+                // Rest client returns an Operation without headers, so we need to create a new response with headers.
+                var headers = new AzureSecurityKeyVaultAdministrationFullBackupHeaders(operation.GetRawResponse());
+                var responseWithHeaders = ResponseWithHeaders.FromValue(headers,operation.GetRawResponse());
+
+                 // Should this return a KeyVaultBackupOperation?
```

Comment:
If we've got a question here, then it would seem we're not ready to merge.   Are you asking if this can/should be `null` or questioning if the type is correct?

Reply:
Oops, that was a note for me while working on this. The answer is yes, the PreBackup and PreRestore operations return the same type as the full Backup and Restore ones.

I'm going to remove this comment.